controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -944,118 +944,158 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
 
     private CompletableFuture<Map<Long, Long>> getTruncationStreamCutBySizeLimit(String scope, String stream, OperationContext context, RetentionPolicy policy,
                                                                                  RetentionSet retentionSet, Map<Long, Long> lowerBound) {
-        // if the lowerbound on subscribers streamcuts satisfies the policy size bound, then return it. 
-        // else return the stream cut that satisfies maximum bound on size. 
         // 1. if lowerbound.size < max and lowerbound.size > min truncate at lowerbound
-        // 2. if lowerbound.size < min, truncate at max irrespective of if lowerbound overlaps with max or not. 
+        // 2. if lowerbound.size < min, truncate at streamcut behind lowerbound that satisfies the policy. (we look at max and below).
         // 3. if lowerbound.size > max, truncate at max
         long currentSize = retentionSet.getLatest().getRecordingSize();
+        Map.Entry<StreamCutReferenceRecord, StreamCutReferenceRecord> limits = getBoundStreamCuts(policy, retentionSet,
```

Comment:
It would be good to explain what this line of code does or have atleast have some  method level comments on `getBoundStreamCuts` explaining what that method takes, computes and returns.

Reply:
I now understand that `getBoundStreamCuts` returns stream-cuts that satisfy the **min** and **max** bounds for the Stream, but it is necessary to explain in the javadoc **how** we arrive at those. It is important to clarify that we calculate all sizes from the current tail of the Stream. While this may be mentioned in the design & user documentation having it in the code keeps the code easier to understand.
For example, If the min limit is 10 GB, there may be stream-cuts, truncating at which will give us sizes of 9.8 GB or 12GB... but here we choose "atleast 10GB" so we select the stream-cut that retains 12GB, though 9.8 GB is _closer_ to the min limit.
However for max, if the max limit is say 50 GB and we have stream-cuts that would keep 45 GB and 51 GB, we choose the one with 45GB because max is "atmost", regardless of the fact that the Stream-cut retaining 51G is closer to the expected max.